algosdk/v2client/algod.py

```diff
@@ -515,9 +581,10 @@ def _specify_round_string(block, round_num):
         round_num (int): user specified variable
     """
 
-    if block is not None and round_num is not None:
-        raise error.OverspecifiedRoundError
-    elif block is not None:
-        return str(block)
-    elif round_num is not None:
+    if not (block is None or round_num is None):
+        raise error.OverspecifiedRoundError()
+
+    if round_num is not None:
         return str(round_num)
+
+    return str(block)
```

Comment:
the prior version returned None if block was None and round_number was None, now you are returning str (None).

Reply:
Here I'm going to push back a bit more.  I argue that the current logic is misleading. It's actually impossible to return `None`. Why?

return `None` meant that:

```python
not (block is not None and round_num is not None)
and not (block is not None)
and not (round_num is not None)
```
That's equivalant to:
```python
not (block is not None and round_num is not None)
and block is None
and round_num is None
```
which is a contradiction.

Therefore, we should not confuse the reader into believing that `None` is possible and that's why I want to make the change

Reply:
ah ok.  I figured that might be true but I didn't turn the crank on it.